src/Xunit.Combinatorial/CombinatorialRangeAttribute.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Andrew Arnott. All rights reserved. Licensed under the Ms-PL.
+
+namespace Xunit
+{
+    using System;
+
+    /// <summary>
+    /// Specifies which range of values for this parameter should be used for running the test method.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public class CombinatorialRangeAttribute : Attribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CombinatorialRangeAttribute"/> class.
+        /// </summary>
+        /// <param name="from">The value at the beginning of the range.</param>
+        /// <param name="count">The quantity of consecutive integer values to include.</param>
+        public CombinatorialRangeAttribute(int from, int count)
+        {
+            if (count < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
+
+            object[] values = new object[count];
+            for (int i = 0; i < count; i++)
+            {
+                values[i] = from + i;
+            }
+
+            this.Values = values;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CombinatorialRangeAttribute"/> class.
+        /// </summary>
+        /// <param name="from">The value at the beginning of the range.</param>
+        /// <param name="to">The value at the end of the range.</param>
+        /// <param name="step">The number of integers to step for each value in result.</param>
+        public CombinatorialRangeAttribute(int from, int to, int step)
+        {
+            // to cannot be less than from
+            if (to < from)
+            {
+                throw new ArgumentOutOfRangeException(nameof(to));
+            }
+
+            // step must be a positive integer
+            if (step < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(step));
+            }
```

Comment:
While these are reasonable constraints, they aren't documented. Can you add to the doc comments for these parameters stating that `from <= to` and `step >= 1`?
Also please document what happens if `from == to`, as well as what happens if `step` jumps *over* `to` rather than stepping to a value equal to it.

Reply:
Docs have been amended. I added explanations of the edge cases in the xml and also added some basic information / a sample test to the readme.